Stripe/UI/STPPaymentCardTextField.m

```diff
@@ -0,0 +1,704 @@
+    
+    self.cvcWidthConstraint = cvcWidthConstraint;
+    [self addConstraint:cvcWidthConstraint];
+    
+    // Make everything be 100% height and vertically centered
+    for (UIView *view in @[self.brandImageView, self.interstitialView, self.dateContainer, self.numberField, self.expirationField, self.cvcField]) {
+        
+        [self addConstraint:[NSLayoutConstraint constraintWithItem:view
+                                                         attribute:NSLayoutAttributeTop
+                                                         relatedBy:NSLayoutRelationEqual
+                                                            toItem:view.superview
+                                                         attribute:NSLayoutAttributeTop
+                                                        multiplier:1.0f
+                                                          constant:0.0f]];
+        
+        [self addConstraint:[NSLayoutConstraint constraintWithItem:view
+                                                         attribute:NSLayoutAttributeHeight
+                                                         relatedBy:NSLayoutRelationEqual
+                                                            toItem:view.superview
+                                                         attribute:NSLayoutAttributeHeight
+                                                        multiplier:1.0f
+                                                          constant:0.0f]];
+        
+    }
+}
+
+typedef void (^STPNumberShrunkCompletionBlock)();
+- (void)setNumberFieldShrunk:(BOOL)shrunk animated:(BOOL)animated
+                  completion:(STPNumberShrunkCompletionBlock)afterCompletion {
+    
+    if (_numberFieldShrunk == shrunk) {
+        if (afterCompletion) {
+            afterCompletion();
+        }
+        return;
+    }
+    
+    _numberFieldShrunk = shrunk;
+    
+    CGFloat nonFragmentWidth = [self widthForCardNumber:[self.viewModel numberWithoutLastDigits]] - 16;
+    CGFloat fragmentWidth = self.numberWidthConstraint.constant - nonFragmentWidth;
+    
+    if (shrunk) {
+        self.dateContainerLeftConstraint.constant = fragmentWidth + 10.0f;
+        [self layoutSubviews];
+    }
+    
+    void (^animations)() = ^void() {
+        self.numberLeftConstraint.constant = shrunk ? -nonFragmentWidth : 0;
+        for (UIView *view in @[self.expirationField, self.cvcField]) {
+            view.alpha = 1.0f * shrunk;
+        }
+        [self layoutSubviews];
+    };
+    
+    void (^completion)(BOOL) = ^void(__unused BOOL succeeded) {
+        if (!shrunk) {
+            self.dateContainerLeftConstraint.constant = 10.0f;
+            [self layoutSubviews];
+        }
+        if (afterCompletion) {
+            afterCompletion();
+        }
+    };
+    
+    FAUXPAS_IGNORED_IN_METHOD(APIAvailability);
+    NSTimeInterval duration = animated * 0.3;
```

Comment:
not super necessary, but this animation duration is a good candidate for a `UIAppearanceSelector`